src/base/misc/disks.c

```diff
@@ -837,20 +838,55 @@ static void image_auto(struct disk *dp)
     leavedos(19);
   }
   lseek(dp->fdesc, 0, SEEK_SET);
-  if (RPT_SYSCALL(read(dp->fdesc, sect, sizeof(sect))) != sizeof(sect)) {
-    error("could not read full header in image_init\n");
+  if (RPT_SYSCALL(read(dp->fdesc, &mbr, sizeof(mbr))) != sizeof(mbr)) {
+    error("could not read first sector as MBR in image_init\n");
+    leavedos(19);
+  }
+  lseek(dp->fdesc, 0, SEEK_SET);
+  if (RPT_SYSCALL(read(dp->fdesc, &vbr, sizeof(vbr))) != sizeof(vbr)) {
```

Comment:
Reading twice is wacky.
How about using a mbr/vbr union?

Reply:
Even better, raw_buf/mbr/vbr union.

Reply:
done

Reply:
Where?

Reply:
You're too fast! I wrote the comment before I pushed it. I also added the dosemu header to the union so that I could remove its corresponding read too.